File: langs/php/src/LoadOptions.php
<?php

declare(strict_types=1);

namespace Dotlyte;

/**
 * Load options for Dotlyte::load() (v2).
 */
final class LoadOptions
{
    /**
     * @param string[]|null $files           Explicit files to load
     * @param string|null   $prefix          Env var prefix to strip
     * @param array<string, mixed> $defaults Default values
     * @param string[]|null $sources         Custom source order
     * @param string|null   $env             Environment name
     * @param array<string, SchemaRule>|null $schema  Validation schema
     * @param bool          $strict          Reject unknown keys
     * @param bool          $interpolateVars Enable ${VAR} interpolation
     * @param array<string, mixed> $overrides Override values (highest priority)
     * @param bool          $debug           Enable debug output
     * @param bool          $findUp          Walk up dirs to find config files
     * @param string[]      $rootMarkers     Root directory markers
     * @param string|null   $cwd             Working directory override
     * @param bool          $allowAllEnvVars Import all env vars without filtering
     */
    public function __construct(
        public readonly ?array $files = null,
        public readonly ?string $prefix = null,
        public readonly array $defaults = [],
        public readonly ?array $sources = null,
        public readonly ?string $env = null,
        public readonly ?array $schema = null,
        public readonly bool $strict = false,
        public readonly bool $interpolateVars = true,
        public readonly array $overrides = [],
        public readonly bool $debug = false,
        public readonly bool $findUp = false,
        public readonly array $rootMarkers = [
            '.git', '.hg', 'package.json', 'composer.json',
            'go.mod', 'Cargo.toml', 'pyproject.toml', '.dotlyte',
        ],
        public readonly ?string $cwd = null,
        public readonly bool $allowAllEnvVars = false,
    ) {}
}

File: langs/dotnet/src/Dotlyte/LoadOptions.cs
namespace Dotlyte;

/// <summary>
/// Load options for DotlyteLoader.Load() (v2).
/// </summary>
public sealed class LoadOptions
{
    /// <summary>Explicit files to load.</summary>
    public string[]? Files { get; init; }

    /// <summary>Environment variable prefix to strip.</summary>
    public string? Prefix { get; init; }

    /// <summary>Default values (lowest priority).</summary>
    public Dictionary<string, object?> Defaults { get; init; } = new();

    /// <summary>Custom source order.</summary>
    public string[]? Sources { get; init; }

    /// <summary>Environment name (e.g., "production").</summary>
    public string? Env { get; init; }

    /// <summary>Schema rules for validation.</summary>
    public Dictionary<string, SchemaRule>? Schema { get; init; }

    /// <summary>Enable strict mode (reject unknown keys).</summary>
    public bool Strict { get; init; }

    /// <summary>Enable variable interpolation (default: true).</summary>
    public bool InterpolateVars { get; init; } = true;

    /// <summary>Override values (highest priority).</summary>
    public Dictionary<string, object?> Overrides { get; init; } = new();

    /// <summary>Enable debug logging.</summary>
    public bool Debug { get; init; }

    /// <summary>Walk up directories to find root marker.</summary>
    public bool FindUp { get; init; }

    /// <summary>Root marker files/dirs for FindUp.</summary>
    public string[] RootMarkers { get; init; } = [".git", "package.json", "go.mod", "Cargo.toml", ".dotlyte"];

    /// <summary>Working directory override.</summary>
    public string? Cwd { get; init; }

    /// <summary>Allow all environment variables (bypass system blocklist).</summary>
    public bool AllowAllEnvVars { get; init; }
}
